packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +270,43 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validateIconsPurpose = (icons: ManifestImageResource[], resource: string, getLocation: JSONLocationFunction) => {
+            for (let iconIndex = 0; iconIndex < icons.length; iconIndex++) {
+                const icon = icons[iconIndex];
+
+                if (!icon.purpose) {
+                    continue;
+                }
+
+                const purposes = icon.purpose.split(' ').filter((purpose) => {
+                    return purpose;
+                });
+                const purposesSet = new Set(purposes);
+
+                for (const purpose of purposesSet) {
+                    if (!validPurposes.includes(purpose)) {
+                        context.report(resource, getMessage('iconPurposeInvalid', context.language, purpose), {
+                            location: getLocation(`icons[${iconIndex}].purpose`),
+                            severity: Severity.warning
+                        });
+                    }
+
+                    const purposeIndex = purposes.indexOf(purpose);
+
+                    if (purposeIndex >= 0) {
+                        purposes.splice(purposeIndex, 1);
+                    }
+                }
+
+                if (purposes.length > 0) {
```

Comment:
It took me a bit to figure out what you were doing in here and why `purposes.length` had to be 0 by line 301.

What do you think of instead doing this:

```suggestion                    
                }

                if (purposes.length !== purposesSet.size) {
```

Reply:
That was my first solution, but you can't know the duplicated items with that code. Maybe I can add a few comments?

Reply:
I think it might be we are doing things in 2 different places.

What about filtering/finding duplicated in the same place?

```ts
if (purposes.length !== purposesSet.size) {
  // Check if it's duplicated here and report
}
```

That way if we want to refactor that code into another function we can. We might have to do an extra loop but it should be negligible and make it better to read/maintain.

Reply:
done